plugins/modules/source_control/gitlab/gitlab_project_variable.py

```diff
@@ -303,9 +397,29 @@ def main():
 
     this_gitlab = GitlabProjectVariables(module=module, gitlab_instance=gitlab_instance)
 
-    change, return_value = native_python_main(this_gitlab, purge, var_list, state, module)
+    change, raw_return_value, before, after = native_python_main(this_gitlab, purge, variables, state, module)
+
+    # postprocessing
+    for item in after:
+        item.pop('project_id')
+        item['name'] = item.pop('key')
+    for item in before:
+        item.pop('project_id')
+        item['name'] = item.pop('key')
+
+    diff = dict(
+        before=yaml.safe_dump(before),
+        after=yaml.safe_dump(after)
+    )
```

Comment:
Pre-formatting diffs is not a good idea. It's better to let the callback plugin do this. Also there's no reason why users should have PyYAML installed, and this code will crash if it isn't installed.
```suggestion
    diff = dict(
        before=before,
        after=after,
    )
```